tests/e2e/GetMatchingContextsTest.php

```diff
@@ -1,40 +1,89 @@
-            [['john_doe', 'contributor2'], 5, ['http://site-ecologique.fr', 'http://site-ecologique-et-politique.fr', 'http://random-site.fr', 'http://expired.fr', 'http://siteecologique.fr']],
-            [['john_doe'], 3, ['http://site-ecologique.fr', 'http://random-site.fr', 'http://siteecologique.fr']],
-            [['contributor2'], 2, ['http://site-ecologique-et-politique.fr', 'http://expired.fr']],
-        ];
+        $payload = $this->makeApiRequest('/api/v3/matchingcontexts');
+
+        self::assertCount(8, $payload);
+
+        $this->assertMatchingContextsAllHaveValidNoticeUrls($payload);
+
+        self::assertEqualsCanonicalizing([
+            '(dismoi\.io)/aide',
+            '(dismoi\.io)/aide',
+            'http://site-ecologique.fr',
+            'http://random-site.fr',
+            'http://site-ecologique-et-politique.fr',
+            'http://expired.fr',
+            "(duckduckgo\.com|www\.bing\.com|www\.google\.fr|www\.qwant\.com|www\.yahoo\.com|first\.domainname\.fr|second\.domainname\.fr)/superexample",
+            'http://siteecologique.fr'
+        ], $this->pluckUrlRegexes($payload));
+
+        $matchingXPath = $this->referenceRepository->getReference('mc_matching_xpath');
+        $aideDisMoi = ArrayHelper::find($payload, function ($matchingContext) use ($matchingXPath) {
+            return $matchingContext['id'] === $matchingXPath->getId();
+        });
+        self::assertEquals("//text()[contains(.,'Aide')]", $aideDisMoi['xpath']);
+
+        $nonMatchingXPath = $this->referenceRepository->getReference('mc_non_matching_xpath');
+        $noFuture = ArrayHelper::find($payload, function ($matchingContext) use ($nonMatchingXPath) {
+            return $matchingContext['id'] === $nonMatchingXPath->getId();
+        });
+        self::assertEquals("//text()[contains(.,'No future')]", $noFuture['xpath']);
     }
 
-    /**
-     * @dataProvider getMatchingContextsData
-     */
-    public function testGetMatchingContexts(?array $contributors, int $count, array $urlRegexes): void
+    public function testGetMatchingContextsForOneContributor(): void
     {
-        $url = '/api/v3/matchingcontexts';
-        if ($contributors) {
-            $url .= '?'.implode('&', array_map(function ($contributorReference) {
-                return 'contributors[]='.$this->referenceRepository->getReference($contributorReference)->getId();
-            }, $contributors));
-        }
-        $payload = $this->makeApiRequest($url);
+        $johnDoe = $this->referenceRepository->getReference('john_doe');
+
+        $payload = $this->makeApiRequest('/api/v3/matchingcontexts?contributors[]='.$johnDoe->getId());
 
-        self::assertCount($count, $payload);
+        self::assertCount(3, $payload);
```

Comment:
Could we add a help message to explicit what's tested here? If possible in the test itself, or in comment otherwise.

Reply:
Or maybe we should simply not test the number of occurrences and only the shape:
- is it an array?
- is it an array of `Something` (with defined keys)?

Reply:
Done